src/components/views/rooms/NotificationBadge/StatelessNotificationBadge.tsx

```diff
@@ -61,10 +61,12 @@ export const StatelessNotificationBadge = forwardRef<HTMLDivElement, XOR<Props,
             mx_NotificationBadge_visible: isEmptyBadge || knocked ? true : hasUnreadCount,
             mx_NotificationBadge_level_notification: level == NotificationLevel.Notification,
             mx_NotificationBadge_level_highlight: level >= NotificationLevel.Highlight,
-            mx_NotificationBadge_dot: (isEmptyBadge && !knocked) || type === "dot",
+            mx_NotificationBadge_dot: type === "dot" || (!knocked && level <= NotificationLevel.Activity),
             mx_NotificationBadge_knocked: knocked,
-            mx_NotificationBadge_2char: type === "badge" && symbol && symbol.length > 0 && symbol.length < 3,
-            mx_NotificationBadge_3char: type === "badge" && symbol && symbol.length > 2,
+            mx_NotificationBadge_2char:
+                type === "badge" && level > NotificationLevel.Activity && (!symbol || symbol.length < 3),
+            mx_NotificationBadge_3char:
+                type === "badge" && level > NotificationLevel.Activity && symbol && symbol.length > 2,
```

Comment:
can you walk me though what's going on with these changes?

Reply:
Ah yes, so this is actually just about changing when the badge becomes a dot: previously, having no content in the badge would have made it a dot, but for mark as unread we want it to be the size of a badge, like a normal notification, but of course it can't have a number. This just allows that.

Reply:
so... we want a badge the size of a dot?

I've got to say, I'm finding this logic completely labyrinthine, and it's getting worse. Is it expected that  more than one of `mx_NotificationBadge_dot`, `mx_NotificationBadge_knocked`, `mx_NotificationBadge_2char` etc can be set at the same time? if not, surely it needs rewriting in a way that makes that clear, rather than having 15 different conditions, each of which compare  `level` and `NotificationLevel.Activity`.

Reply:
also, is this labyrinthine logic tested in any of the jest tests? I couldn't see anything about it, but maybe I'm missing it.

Reply:
In this case, more a dot the size of a badge, although that's the same thing as a badge with no contents. I've rewritten the logic a little in ae809dc7ab and renamed the param to more what it actually does, ie. sometimes it forces the dot, sometimes it lets the component decide. Hopefully that's clearer.